### PR TITLE
XDAI support using JSON-RPC provider

### LINKED ISSUE #15: App should also work on xDAI
Since our second customer is already on xdai, we should enable the app also on xDAI

If it is easier, we can also deploy it on a different url for different networks

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #15 

I got some errors in the console related to json-rpc provider and a tx reverted, but the auction seems like was initiated correctly.

### CODE REVIEW

**Comment by josojo** [comment]
Niice, code looks good. I will test it throughly tomorrrow

**Comment by josojo** [comment]
For me the build button is not enabled...
image
Would be great to get a fix soon

**Comment by josojo** [comment]
After reloading, I was able to create a tx. but it failed...

Looking at it, it had a safeTxGas amount of 0. On rinkeby they are different:
```
4 | safeTxGas | uint256 | 388617
```

**Comment by josojo** [comment]
I asked internally and got the follwing answers:

Lukas Schor  5 minutes ago
Might be related to this: https://github.com/gnosis/safe-react/issues/1932 (included in the next Sprint)
tschubotztschubotz
#1932 Gas estimation on xDai fails always
Note: this seems to be an RPC endpoint issue. Everything works with https://rpc.xdaichain.com/ but not with https://rpc.xdaichain.com ..
Title/Description
Gas estimation on xDai fails always
Environment
• Browser: Firefox
• Safe: https://xdai.gnosis-safe.io/app/#/safes/0x82FC6069FB9290F98c6ae02b6cfb7B6aB0ab97bf/transactions
• Wallet: Metamask (rpc.xdaichain.com)
• App Version: 3.1.0
• Environment:
Show more
Labels
Bug :bug:, Major
<https://github.com/gnosis/safe-react|gnosis/safe-react>gnosis/safe-react | Feb 19th | Added by GitHub (98 kB)
https://camo.githubusercontent.com/d7d91309d925f1ec324c74d3e47af0d7701ffea5365095f90004abfa7f881308/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3537616464346665653430653537313462313664343936392f33393562616132382d306137302d343834392d613830622d366338656231333938356236
:+1:
1





rimeissner  4 minutes ago
for a quick test you can just try to up the gas limit of the ethereum transaction to something really high (e.g. 6mio)

**Comment by josojo** [comment]
In even another attempt, its working: https://blockscout.com/xdai/mainnet/tx/0x78d384c00403577f65188b27d8b87bb9417753a652e66bf82cd30bb0859ba840/token-transfers

I think we should test it some more times and investigate when its working and when not... But its must be due to some gas estimations

File: src/networkConfig.ts [hunk 1]
```diff
@@ -18,3 +20,15 @@ export const EASY_AUCTION_ADDRESSES: EasyAuctionAdresses = {
...
+
+const INFURA_ID = process.env.REACT_APP_INFURA_ID
+if (!INFURA_ID) {
+  throw new Error('REACT_APP_INFURA_ID not provided')
+}
+
+const XDAI_RPC = process.env.REACT_APP_XDAI_RPC || 'https://rpc.xdaichain.com'
+
+export const getProvider = (network: Networks) => {
+  const rpc = network === 'XDAI' ? XDAI_RPC : `https://${network}.infura.io/v3/${INFURA_ID}`
```
Comment (hpmaxi) [inline src/networkConfig.ts:32]: @josojo this should make it possible to use another rpc if is needed.

**Review by josojo — APPROVED** [review]
I think this PR is ready to be merged:

I could no longer get the error while testing
image
I think the safe made some improvements in their backend code and hence, its fixed. 

And we can always manually adjust the gasLimits

**Pull request merged**